build/azure-pipelines/win32/sdl-scan-win32.yml

```diff
@@ -26,40 +25,34 @@ steps:
       KeyVaultName: vscode-build-secrets
       SecretsFilter: "github-distro-mixin-password"
 
+  - powershell: node build/setup-npm-registry.js $env:NPM_REGISTRY
+    condition: and(succeeded(), ne(variables['NPM_REGISTRY'], 'none'))
+    displayName: Setup NPM Registry
+
   - powershell: |
       . build/azure-pipelines/win32/exec.ps1
       $ErrorActionPreference = "Stop"
-      exec { npm config set registry "${{ parameters.NPM_REGISTRY }}" --location=project }
+      exec { npm config set registry "$env:NPM_REGISTRY" --location=project }
       # npm >v7 deprecated the `always-auth` config option, refs npm/cli@72a7eeb
       # following is a workaround for yarn to send authorization header
       # for GET requests to the registry.
       exec { Add-Content -Path .npmrc -Value "always-auth=true" }
-      exec { yarn config set registry "${{ parameters.NPM_REGISTRY }}" }
-    condition: and(succeeded(), ne(variables.NODE_MODULES_RESTORED, 'true'), ne('${{ parameters.NPM_REGISTRY }}', 'none'))
+      exec { yarn config set registry "$env:NPM_REGISTRY" }
+    condition: and(succeeded(), ne(variables.NODE_MODULES_RESTORED, 'true'), ne(variables['NPM_REGISTRY'], 'none'))
```

Comment:
`NODE_MODULES_RESTORED` in this template doesn't seem right as the cache is never used today. Should remove dependency on it until the cache is restored.